build_test/lib/src/in_memory_reader.dart

```diff
@@ -17,19 +17,17 @@ class InMemoryAssetReader implements AssetReader {
       : assets = sourceAssets ?? <AssetId, DatedValue>{};
 
   @override
-  Future<bool> hasInput(AssetId id) {
-    return new Future.value(assets.containsKey(id));
-  }
+  FutureOr<bool> canRead(AssetId id) => assets.containsKey(id);
```

Comment:
Since `hasInput` is just deprecated don't we still need it here (and other impls below)